include/nbl/builtin/hlsl/fft/common/twiddles.hlsl

```diff
@@ -0,0 +1,43 @@
+#ifndef _NBL_BUILTIN_HLSL_FFT_COMMON_TWIDDLES_INCLUDED_
+#define _NBL_BUILTIN_HLSL_FFT_COMMON_TWIDDLES_INCLUDED_
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/complex.hlsl"
+
+#define LOW_TWIDDLE_BITS 6
+#define MID_TWIDDLE_BITS 5
+#define HIGH_TWIDDLE_BITS 4 
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace fft::common {
+
+    template<typename Scalar>
+    complex_t<Scalar> getHigh(uint32_t highIdx); 
+
+    template<typename Scalar>
+    complex_t<Scalar> getMid(uint32_t midIdx); 
+
+    template<typename Scalar>
+    complex_t<Scalar> getLow(uint32_t lowIdx);
+
+    // Define twiddles and getters for each Scalar template specialization
+    #define float_t float32_t
+    #define TYPED_NUMBER(N) NBL_CONCATENATE(N, f) // to add f after floating point numbers and avoid casting warnings and emitting ShaderFloat64 Caps
+    #include <nbl/builtin/hlsl/fft/common/twiddles_impl.hlsl>
+    #undef TYPED_NUMBER
+    #undef float_t 
+
+    #define float_t float64_t
+    #define TYPED_NUMBER(N) N
+    #include <nbl/builtin/hlsl/fft/common/twiddles_impl.hlsl>
+    #undef TYPED_NUMBER
+    #undef float_t
```

Comment:
don't bother precomputing twiddles, did this benchmark in shadertoy
```cpp
const float _cos[32] = float[32](
1.0,0.980785,0.92388,0.83147,0.707107,0.55557,0.382683,0.19509,-4.37114e-08,-0.19509,-0.382684,-0.55557,-0.707107,-0.83147,-0.92388,-0.980785,-1.0,-0.980785,-0.92388,-0.83147,-0.707107,-0.55557,-0.382683,-0.19509,1.19249e-08,0.19509,0.382684,0.55557,0.707107,0.83147,0.92388,0.980785
);

void mainImage( out vec4 fragColor, in vec2 fragCoord )
{
    // Normalized pixel coordinates (from 0 to 1)
    vec2 uv = fragCoord/iResolution.xy;

    // Time varying pixel color
    vec3 col = vec3(0.0);

    const int spins = 2000;
    for (int i=0; i<spins; i++)
    {
        int iarg = i;//+int(fragCoord.x)+int(fragCoord.y);
        float arg = float(iarg)/float(spins);
        float cs = cos(arg*6.2831853);
        col.rg += vec2(_cos[iarg]);
    }
    // Output to screen
    fragColor = vec4(col,1.0);
}
```
perf dropped 50%, it gets worse the bigger the array gets